### DISCUSSION TITLE
Improve the non built-in modules disabling

### DISCUSSION THREAD

**Discussion by hrostyk** [PrestaShop/PrestaShop#32344] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When I press the button <Disable non built-in modules> it does not change its status. And incorrect title on mouse over  <title="Clear cache">. I am not sure if this function works.

![CleanShot 2023-01-08 at 18 15 09@2x](https://user-images.githubusercontent.com/7484618/211207277-4e92817b-8e60-4022-ba02-fb180af06d86.png)

### Expected behavior

If all non-built-in modules are disabled, the button must be disabled.
Also change the title to <title="Disable non built-in modules">

### Steps to reproduce

1. BO
2. Advanced Parameters -> Performance
3. Modules

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

7.4.33

**Comment by hrostyk** [comment]
Sorry, I did not immediately understand how it works. I thought it Disable and Enable all non built-in modules. This button only turns off! Then you need to turn on each module separately. 
If all non built-in modules are disabled, the button must be disable.

**Comment by hibatallahAouadni** [comment]
Hello @hrostyk 

Indeed this button only disable the non-built modules as you can see in the specs #9965
Ping @PrestaShop/product-team could you please explain why this button doesn't turn grey when all non-build modules are disabled 🙏 

Thanks in advance!

**Comment by MatShir** [comment]
> If all non built-in modules are disabled, the button must be disable.

Could be a nice improvement :)

The problem of disabling and enabling all is if you have one module A disabled (bc you don't want to use it for your shop) and one module B enabled. So you disable them all and enable all them back then you will enable all of them regarless of their initial state. I don't think it is optimal for the users. Also, the module manager has a bulk feature to enable modules if you really want to do it in fewer clicks.

I keep the issue as improvement for "To disable the button is all the modules are disabled". Still open to discuss :)

**Comment by hrostyk** [comment]
To continue this topic...

When click "Disable" save state (Enable/Disable) for all modules (e.g. save all module ID of enabled non built-in modules) and change button to "Enable". Then you can implement the function "Enable". If turn on each module separately clear saved module ID's and change button to "Disable". When all modules are disabled... disable button with "Disable".

**Comment by MatShir** [comment]
Sounds a better solution, would you be interested to contribute maybe ?
Meanwhile I can update the specs of the issue.
